cdap-tms/src/main/java/io/cdap/cdap/messaging/subscriber/AbstractMessagingSubscriberService.java

```diff
@@ -144,61 +146,64 @@ protected String processMessages(Iterator<ImmutablePair<String, T>> messages) th
                                                                                               timeBoundMillis);
       MessageTrackingIterator trackingIterator = new MessageTrackingIterator(timeBoundMessages);
       processMessages(context, trackingIterator);
-      String lastMessageId = trackingIterator.getLastMessageId();
-
-      // Persist the message id of the last message being consumed from the iterator
-      if (lastMessageId != null) {
-        storeMessageId(context, lastMessageId);
+      ImmutablePair<String, T> lastConsumer = trackingIterator.getLastComputedMessage();
```

Comment:
since the value of the pair is never used, cleaner for this method just to return the message id.

Reply:
Done